back/app/services/admin_publications_filtering_service.rb

```diff
@@ -51,4 +53,14 @@ class AdminPublicationsFilteringService
   add_filter('top_level_only') do |scope, options|
     [0, '0'].include?(options[:depth]) ? scope.where(depth: 0) : scope
   end
+
+  add_filter('remove_childless_parents') do |scope, options|
+    next scope unless ['true', true, '1'].include? options[:remove_childless_parents]
+
+    parents_of_project_publications_ids = project_publications.where.not(parent_id: nil).map(&:parent_id)
```

Comment:
The `map` will trigger a DB query here.

Reply:
I've removed it, using your suggestion with `select`. Does this mean we should also edit out the existing `map` statements in the first `'remove_not_allowed_parents'` filter?

Reply:
That's up to you 😉

Reply:
I tried replacing the other uses of `map`, but everything in `admin_publications_filtering_spec` failed, and I think I would need to understand `class MockAdminPublicationsTree` before I could hope to understand why. So I'll leave it for another day / ticket.